src/dmd/parse.d

```diff
@@ -4447,6 +4447,31 @@ final class Parser(AST) : Lexer
                         parseAttributes;
 
                     AST.Declaration v;
+                    AST.Dsymbol s;
+
+                    // allow declaring func type with the AliasDeclarationY flavor,
+                    // e.g `alias ident = Type (Parameters);`
+                    Token* tlu = &token;
+                    if (token.value != TOK.function_ &&
+                        token.value != TOK.delegate_ &&
+                        isBasicType(&tlu) && tlu &&
```

Comment:
What about other than basic types?

Reply:
This is a pragmatic choice. There's no lookup function for `Type`.

Reply:
is this handled the same way in other places in the parser?

Reply:
I don't know but I'll will have to do it anyway since the C-style syntax support a full Type (note that the only difference is the random amount of TypeCtors before the BasicType).

Reply:
Actually the type ctors are part of the function type, not the return. So it's okay to just parse a BasicType. Then the type ctors can be placed as postfixes.